plugins/optimization-detective/class-od-html-tag-processor.php

```diff
@@ -317,13 +327,40 @@ public function next_token(): bool {
 				$i = array_search( 'P', $this->open_stack_tags, true );
 				if ( false !== $i ) {
 					array_splice( $this->open_stack_tags, (int) $i );
+					array_splice( $this->open_stack_attributes, (int) $i );
 					array_splice( $this->open_stack_indices, count( $this->open_stack_tags ) + 1 );
 				}
 			}
 
 			$level                   = count( $this->open_stack_tags );
 			$this->open_stack_tags[] = $tag_name;
 
+			/*
+			 * Capture the most stable attribute which can be used to disambiguate an XPath expression when the node index is not appropriate.
+			 * This is used specifically for children of the BODY. The `id` and `role` attributes are most stable (cf. <https://g.co/gemini/share/032edd9063c1>),
+			 * although all Block Themes utilize the 'wp-site-blocks' class name in the root `DIV`.
+			 * TODO: Consider only doing this if $this->open_stack_tags[-2] === 'BODY' and $level === 2 (e.g. we're at a child of /HTML/BODY) since this is the only place we need this.
```

Comment:
- [ ] Add condition to avoid needlessly computing `$attributes` when they will not be used.